### PR TITLE
network: wireguard: allow to start ndisc or radv

### LINKED ISSUE #17380: No router advertisements sent out over wireguard link
**systemd version the issue has been seen with**
246

**Used distribution**
Arch Linux

**Linux kernel version used** (`uname -a`)
Linux 5.9.1-arch1-1

**CPU architecture issue was seen on**
x86_64

**Expected behaviour you didn't see**
Client autoconfiguring ipv6 addresses by using router advertisements sent out over wireguard link by the server.

**Unexpected behaviour you saw**
Router advertisements were not sent.

**Steps to reproduce the problem**
Setup the following configuration:

### Server configuration:
`/etc/systemd/network/20-wg0.netdev`:
```
[NetDev]
Name=wg0
Kind=wireguard

[WireGuard]
PrivateKey=server_private_key
ListenPort=12345

[WireGuardPeer]
PublicKey=client_public_key
AllowedIPs=::/0
```

`/etc/systemd/network/20-wg0.network`:
```
[Match]
Name=wg0

[Link]
Multicast=yes

[Network]
Address=fd00:1234:5678:9abc::1/64
IPv6PrefixDelegation=static

[Address]
Address=fe80::1234:5678:9abc:def0/64
Scope=link

[IPv6PrefixDelegation]
RouterLifetimeSec=0
DNS=fd00:1234:5678:9abc::1
Domains=example.com
DNSLifetimeSec=1800

[IPv6Prefix]
Prefix=fd00:1234:5678:9abc::/64
```

### Client configuration:
`/etc/systemd/network/20-wg0.netdev`
```
[NetDev]
Name=wg0
Kind=wireguard

[WireGuard]
PrivateKey=client_private_key

[WireGuardPeer]
PublicKey=server_public_key
AllowedIPs=::/0
Endpoint=wireguard.example.com:12345
```

`/etc/systemd/network/20-wg0.network`
```
[Match]
Name=wg0
Type=wireguard

[Link]
Multicast=yes

[Network]
IPv6Token=static:::0fed:cba9:8765:4321
IPv6AcceptRA=yes

[Address]
Address=fe80::0fed:cba9:8765:4321/64
Scope=link

[IPv6AcceptRA]
UseDomains=route
```

### PR DESCRIPTION
**Pull request by yuwata**
Fixes #17380.

### CODE REVIEW

File: src/network/networkd-link.c [hunk 1]
```diff
@@ -126,6 +126,43 @@ bool link_ipv6ll_enabled(Link *link) {
...
+                if (!link->network)
+                        return false;
+
+                ORDERED_HASHMAP_FOREACH(a, link->network->addresses_by_section) {
+                        if (a->family != AF_INET6)
+                                continue;
+                        if (in6_addr_is_set(&a->in_addr_peer.in6))
+                                continue;
+                        if (in6_addr_is_link_local(&a->in_addr.in6))
+                                return true;
```
Comment (keszybz) [inline src/network/networkd-link.c:159]: Hmm, we get here only if the address is all 0, so `_is_link_local` can never match??

Comment (yuwata) [reply src/network/networkd-link.c:159]: Hm? `in6_addr_is_set()` is called for the peer address. Or did I misunderstand what you want to say?

Comment (keszybz) [reply src/network/networkd-link.c:159]: Oh, OK, different addresses. NVM.

**Comment by keszybz** [comment]
LGTM.

**Pull request merged**